internal/provider/resource_access_model_test.go

```diff
@@ -123,51 +143,66 @@ func TestAcc_ResourceAccessModel_ErrorWhenUsedWithJAAS(t *testing.T) {
 
 func testAccResourceAccessModelFixedUser() string {
 	return `
+resource "juju_model" "test-model" {
+	name = "test-model"
+}
 resource "juju_access_model" "test" {
   access = "write"
-  model = "foo"
+  model_uuid = juju_model.test-model.uuid
   users = ["bob"]
 }`
 }
 
-func testAccResourceAccessModel(userName, userPassword, modelName, access string) string {
-	return fmt.Sprintf(`
-resource "juju_user" "test-user" {
-  name = %q
-  password = %q
+func testAccResourceAccessModel(userName, userPassword, modelName, access string, useModelUUID bool) string {
+	return internaltesting.GetStringFromTemplateWithData("testAccResourceAccessModel",
+		`resource "juju_user" "test-user" {
+  name = "{{.UserName}}"
+  password = "{{.UserPassword}}"
 }
 
-resource "juju_model" "test-model" {
-  name = %q
+resource "juju_model" "{{.ModelName}}" {
+  name = "{{.ModelName}}"
 }
 
 resource "juju_access_model" "test" {
-  access = %q
-  model = juju_model.test-model.name
+  access = "{{.Access}}"
+  {{- if eq .UseModelUUID false }}
+  model = juju_model.{{.ModelName}}.name
+  {{- else }}
+  model_uuid = juju_model.{{.ModelName}}.uuid
+  {{- end }}
   users = [juju_user.test-user.name]
-}`, userName, userPassword, modelName, access)
+}`, internaltesting.TemplateData{
+			"UserName":     userName,
+			"UserPassword": userPassword,
+			"ModelName":    modelName,
+			"Access":       access,
+			"UseModelUUID": useModelUUID,
+		})
 }
 
-func TestAcc_ResourceAccessModel_Schema_v0_To_v1(t *testing.T) {
+func TestAcc_ResourceAccessModel_UpgradeV0ToV2(t *testing.T) {
 	SkipJAAS(t)
+
 	user1 := acctest.RandomWithPrefix("tfuser1")
 	password1 := acctest.RandomWithPrefix("tf-test-user1")
 	user2 := acctest.RandomWithPrefix("tfuser2")
 	password2 := acctest.RandomWithPrefix("tf-test-user2")
 	modelName := acctest.RandomWithPrefix("tf-access-model")
 
 	resourceName := "juju_access_model.test"
+
 	resource.ParallelTest(t, resource.TestCase{
 		PreCheck: func() { testAccPreCheck(t) },
 		Steps: []resource.TestStep{
 			{
 				ExternalProviders: map[string]resource.ExternalProvider{
 					"juju": {
-						VersionConstraint: "0.21.1",
+						VersionConstraint: TestProviderPreV1Version,
```

Comment:
I would leave this as 0.21.1 because 0.22 will have the access_model resource with a `set` rather than `string`, so if we update `TestProviderPreV1Version` that could break the test. I say "could" because I think the test would still pass but more as a coincidence because the plan syntax doesn't change between `set` and `list`.